File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y
  | dist > 10 = 0
  | dist <= 10 && dist > 5 = 1
  | dist <= 5 && dist > 1 = 5
  | otherwise = 10
    where
    dist = sqrt (((x - 0) ^ 2) + ((y - 0) ^ 2))

File: solutions/cpp/darts/1/darts.cpp
#include "darts.h"

namespace darts {

    int score(float x, float y) {
        float dist_from_center = std::sqrt( std::pow(x, 2) + std::pow(y, 2) );
        if (dist_from_center > 10.0f) {
            return 0;
        } else if (dist_from_center > 5.0f && dist_from_center <= 10.0f) {
            return 1;
        } else if (dist_from_center > 1.0f && dist_from_center <= 5.0f) {
            return 5;
        } else {
            return 10;
        }
    }

}  // namespace darts
